core/src/test/scala/unit/kafka/admin/ReassignPartitionsServiceClientTest.scala

```diff
@@ -0,0 +1,394 @@
+    reset(mockZkClient)
+    reset(mockAdminZkClient)
+    var test3Properties = new Properties
+    val test3PropertiesMap = Map("otherProp1" -> "otherVal1", "property2" -> "origVal2")
+    var test3ExpectedProperties = new Properties()
+    for ((k,v) <- test3PropertiesMap) {
+      test3Properties.setProperty(k, v)
+      test3ExpectedProperties.setProperty(k, v)
+    }
+    for ((k,v) <- configUpdatesMap) {
+      test3ExpectedProperties.setProperty(k, v)
+    }
+
+    expect(mockAdminZkClient.fetchEntityConfig(ConfigType.Broker, testBrokerId.toString)).andReturn(test3Properties)
+    expect(mockAdminZkClient.changeBrokerConfig(Seq(testBrokerId), test3ExpectedProperties))
+    replay(mockZkClient)
+    replay(mockAdminZkClient)
+
+    val serviceTest3 = new ZkClientReassignCommandService(mockZkClient, Some(mockAdminZkClient))
+
+    serviceTest3.UpdateBrokerConfigs(testBrokerId, configUpdatesMap)
+  }
+
+}
+
+/**
+ * Validate the AdminServiceClient used by the ReassignPartitionsCommand.
+ */
+class ReassignPartitionsServiceClientAdminTest {
+  // Define a standardized test cluster
+  val brokerIds = List(1, 4, 6, 7)
+  val portBase = 9000
+  // Try with a set of brokers where some have rack info and some don't
+  val brokerList = brokerIds.map { id => new Node(id, s"broker-$id", portBase, if (id % 2 == 0) { "" } else { s"rack-$id" }) }
+  // Topic test 1 has 2 partitions, 3RF, one partition is UR
+  // XXX: MockAdminClient doesn't allow partitions with different replica sets
+  val test1Partitions = List(new TopicPartitionInfo(1, brokerList(0), List(brokerList(0), brokerList(1), brokerList(2)).asJava,
+    List(brokerList(0), brokerList(1)).asJava),
+    new TopicPartitionInfo(2, brokerList(1), List(brokerList(0), brokerList(1), brokerList(2)).asJava,
+      List(brokerList(0), brokerList(1), brokerList(2)).asJava)).asJava
+
+  // Topic test2 has just 1 partition, RF 3, 1 UR partition
+  val test2Partitions = List(new TopicPartitionInfo(1, brokerList(3), List(brokerList(2), brokerList(3), brokerList(0)).asJava,
+    List(brokerList(3), brokerList(0)).asJava)).asJava
+
+  // Topic test3 has 3 partitions, RF3
+  val test3Partitions =  List(new TopicPartitionInfo(1, brokerList(0), List(brokerList(0), brokerList(1), brokerList(2)).asJava,
+    List(brokerList(0), brokerList(1)).asJava),
+    new TopicPartitionInfo(2, brokerList(1), List(brokerList(0), brokerList(1), brokerList(2)).asJava,
+      List(brokerList(0), brokerList(1), brokerList(2)).asJava),
+    new TopicPartitionInfo(3, brokerList(1), List(brokerList(0), brokerList(1), brokerList(2)).asJava,
+      List(brokerList(0), brokerList(1), brokerList(2)).asJava)).asJava
+
+
+  /**
+   * Set up the standardized test cluster.
+   */
+  def setUpTestCluster() : Admin = {
+    // Note the non-sequential, non-zero-based IDs.
+    val mockAdmin = new MockAdminClient(brokerList.asJava, brokerList(2))
+
+    mockAdmin.addTopic(false, "test1", test1Partitions, Collections.emptyMap())
```

Comment:
Could we move the `test1` and etc topic names to variables? That way we better know what topics the admin client from `setUpTestCluster()`  supports, rather than flying strings